galsim/chromatic.py

```diff
@@ -1359,3 +1376,286 @@ def withScaledFlux(self, flux_ratio):
+        of interpolated images at each wavelength to carry out the integration, only carrying out
+        the GSObject instantation and convolution with pixel response at the very end.
+
+        Currently there is only one image integrator for direct integration of the interpolated
+        images, so there is no keyword argument to select one, unlike for
+        ChromaticObject.drawImage().
+
+        @param bandpass         A Bandpass object representing the filter against which to
+                                integrate.
+        @param force_eval       Option to force image rendering to take the brute-force evaluation
+                                of images instead of relying on interpolation.  [default: False]
+        @param image            Optionally, the Image to draw onto.  (See GSObject.drawImage()
+                                for details.)  [default: None]
+        @param **kwargs         For all other kwarg options, see GSObject.drawImage()
+
+        @returns the drawn Image.
+        """
+        if (self.waves is None) or force_eval:
+            return ChromaticObject.drawImage(self, bandpass, image=image, **kwargs)
+
+        # setup output image (semi-arbitrarily using the bandpass effective wavelength)
+        prof0 = self.evaluateAtWavelength(bandpass.effective_wavelength)
+        image = prof0.drawImage(image=image, setup_only=True, **kwargs)
+        # Remove from kwargs anything that is only used for setting up image:
+        if 'dtype' in kwargs: kwargs.pop('dtype')
+        if 'scale' in kwargs: kwargs.pop('scale')
+        if 'wcs' in kwargs: kwargs.pop('wcs')
+
+        # determine combined self.wave_list and bandpass.wave_list
+        wave_list = self._getCombinedWaveList(bandpass)
+
+        # decide on integrator
+        integrator = galsim.integ.DirectSampleIntegrator(galsim.integ.midpt)
+
+        # merge self.wave_list into bandpass.wave_list since we are using a sampling integrator
+        bandpass = galsim.Bandpass(galsim.LookupTable(wave_list, bandpass(wave_list),
+                                                      interpolant='linear'))
+
+        integral, stepk, maxk = integrator(self._imageAtWavelength, bandpass)
+        # For now, pretend we have no information about the maxk and stepk that should be used.
+        int_im = galsim.InterpolatedImage(integral, _force_stepk=stepk, _force_maxk=maxk)
+
+        # For performance profiling, store the number of evaluations used for the last integration
+        # performed.
+        self._last_n_eval = integrator.last_n_eval
+
+        # Apply integral to the initial image appropriately.  This will naturally work properly and
+        # take into account the supplied value of `add_to_image`, which will be included in kwargs.
+        image = int_im.drawImage(image=image, **kwargs)
+        return image
+
+    def withScaledFlux(self, flux_ratio):
+        ret = self.copy()
+        ret.SED *= flux_ratio
```

Comment:
I think you're right.

Now that I've gone back to the traditional usage, where if this is a galaxy it'll have a proper SED and it only has ```lambda w:1.0``` if it's a PSF, then does that matter?  (I don't think people would be using ```withScaledFlux``` if it's a PSF, would they?)